src/worker/wrapTryCatchAroundAction.js

```diff
@@ -0,0 +1,32 @@
+import { compact } from "lodash-es";
+
+/**
+ * Wraps the given action in a try-catch that logs the error message.
+ *
+ * @param {Logger}   logger                  The logger instance to log with.
+ * @param {Function} action                  The action to safely run.
+ * @param {string}   [errorMessagePrefix=""] The prefix of the error message.
+ *
+ * @returns {Function} The wrapped action.
+ */
+export default function wrapTryCatchAroundAction( logger, action, errorMessagePrefix = "" ) {
+	return ( ...args ) => {
+		try {
+			return action( ...args );
+		} catch ( error ) {
+			let errorMessage = [ errorMessagePrefix ];
```

Comment:
By using this, compact would become unnecessary 🙂.
```suggestion
			let errorMessage = errorMessagePrefix ? [ errorMessagePrefix ] : [];
```

Reply:
Good point! 😄